### PR TITLE
docs: Add det pach tutorial

### PR DESCRIPTION
**Pull request by tara-det-ai**
## Description

As a user, I want to understand how Determined and Pachyderm can be used together to build a data pipeline and train a model. I want an easy-to-follow example with step-by-step instructions.

## Example

https://github.com/pachyderm/examples/tree/master/determined-pachyderm-batch-inferencing

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: docs/tutorials/pachyderm-cat-dog.rst [hunk 1]
```diff
@@ -0,0 +1,413 @@
...
+
+      .. code:: bash
+
+         pachctl version
+
+      You should see a configuration summary the displays the versions for ``pachctl`` and
+      ``pachd``.
+
+****************************************
+ Create a Project for Batch Inferencing
```
Comment (MikhailKardash) [inline docs/tutorials/pachyderm-cat-dog.rst:106]: Maybe "Pachyderm Project"? Determined also has a Project term.

File: docs/tutorials/pachyderm-cat-dog.rst [hunk 2]
```diff
@@ -0,0 +1,413 @@
...
+
+   pachctl config update context --project batch-inference-1
+
+By setting up a dedicated project, you ensure that all related repos and pipelines are organized
+cohesively. This encapsulation makes it easier to manage batch inferencing workflows in the future.
+
+You are now ready to create a project repo.
+
+********************************
+ Create Repos for Training Data
```
Comment (MikhailKardash) [inline docs/tutorials/pachyderm-cat-dog.rst:130]: Pachyderm Repos

**Comment by MikhailKardash** [comment]
Only have a minor gripe with the terms "Project" and "Repo", since users not familiar with Pachyderm should probably know that these are features of Pachyderm, and not features of Determined/GitHub. I'm in favor of introducing them as "Pachyderm Project" and "Pachyderm Repo" at least initially, in the headings. To a lesser extent, this also applies to the term Pipeline, but Pipelines are clearly a feature of Pachyderm here.

File: docs/tutorials/pachyderm-cat-dog.rst (hunk 1 above)
Comment (tara-det-ai) [reply docs/tutorials/pachyderm-cat-dog.rst:106]: good point!

**Review by lbliii — APPROVED** [review]

**Review by MikhailKardash — APPROVED** [review]

**Pull request merged**